traj_point_data[] objs
